core/src/components/radio-group/test/validation/index.html

```diff
@@ -0,0 +1,195 @@
+      }
+
+      h2 {
+        font-size: 12px;
+        font-weight: normal;
+
+        color: var(--ion-color-step-600);
+
+        margin-top: 10px;
+        margin-bottom: 5px;
+      }
+
+      .validation-info {
+        margin: 20px;
+        padding: 10px;
+        background: var(--ion-color-light);
+        border-radius: 4px;
+      }
+    </style>
+  </head>
+
+  <body>
+    <ion-app>
+      <ion-header>
+        <ion-toolbar>
+          <ion-title>Radio Group - Validation Test</ion-title>
+        </ion-toolbar>
+      </ion-header>
+
+      <ion-content class="ion-padding">
+        <div class="validation-info">
+          <h2>Screen Reader Testing Instructions:</h2>
+          <ol>
+            <li>Enable your screen reader (VoiceOver, NVDA, JAWS, etc.)</li>
+            <li>Tab through the form fields</li>
+            <li>When you tab away from an empty required field, the error should be announced immediately</li>
+            <li>The error text should be announced BEFORE the next field is announced</li>
+            <li>Test in Chrome, Safari, and Firefox to verify consistent behavior</li>
+          </ol>
+        </div>
+
+        <div class="grid">
+          <div>
+            <h2>Required Field</h2>
+            <ion-radio-group
+              id="fruits-radio-group"
+              helper-text="You must select one to continue"
+              error-text="Please select a fruit"
+              allow-empty-selection="true"
+              required
+            >
+              <ion-radio value="grapes">Grapes</ion-radio><br />
+              <ion-radio value="strawberries">Strawberries</ion-radio>
+            </ion-radio-group>
+          </div>
+
+          <div>
+            <h2>Optional Field (No Validation)</h2>
+            <ion-radio-group
+              id="optional-radio-group"
+              helper-text="You can skip this field"
+              allow-empty-selection="true"
+              required
+            >
+              <ion-radio value="cucumbers">Cucumbers</ion-radio><br />
+              <ion-radio value="tomatoes">Tomatoes</ion-radio>
+            </ion-radio-group>
+          </div>
+        </div>
+
+        <div class="ion-padding">
+          <ion-button id="submit-btn" expand="block" disabled>Submit Form</ion-button>
+          <ion-button id="reset-btn" expand="block" fill="outline">Reset Form</ion-button>
+        </div>
+      </ion-content>
+    </ion-app>
+
+    <script>
+      // Simple validation logic
+      const radioGroups = document.querySelectorAll('ion-radio-group');
+      const submitBtn = document.getElementById('submit-btn');
+      const resetBtn = document.getElementById('reset-btn');
+
+      // Track which fields have been touched
+      const touchedFields = new Set();
+
+      // Validation functions
+      const validators = {
+        'fruits-radio-group': (value) => {
+          return value !== undefined;
+        },
+        'optional-checkbox': () => true, // Always valid
```

Comment:
I think this is a copy-paste issue from checkbox, `optional-checkbox` doesn't exist here (in radio group)

Reply:
https://github.com/ionic-team/ionic-framework/pull/30714/commits/ec28127c12b11b1b58e8dd3406175f927524420d